src/routers/datatypeRouters/edges/variants_phenotypes.ts

```diff
@@ -150,90 +187,47 @@ async function getHyperedgeFromVariantQuery (input: paramsFormatType): Promise<a
-        ${getDBReturnStatements(variantPhenotypeToStudy).replaceAll('record', 'edgeRecord')}
-      }
-      `
-    } else {
-      query = `
-      LET primarySources = (
-        FOR record IN ${variantSchema.db_collection_name} ${queryOptions}
-        FILTER ${variantFilters}
-        RETURN record._id
-      )
-
-      LET primaryEdges = (
-        FOR record IN ${variantToPhenotypeSchema.db_collection_name}
-        FILTER record._from IN primarySources ${phenotypeFilter}
-        RETURN record._id
-      )
+    query = `
+    LET primaryEdges = (
+      FOR record IN ${variantToPhenotypeSchema.db_collection_name as string}
+      FILTER record._from IN ['${variantIDs.join('\', \'')}']  ${phenotypeFilter}
+      RETURN record._id
+    )
 
-      FOR edgeRecord IN ${variantPhenotypeToStudy.db_collection_name}
-      FILTER edgeRecord._from IN primaryEdges ${hyperEdgeFilter.replaceAll('record', 'edgeRecord')}
-      SORT '_key'
+    FOR edgeRecord IN ${variantPhenotypeToStudy.db_collection_name as string}
+    FILTER edgeRecord._from IN primaryEdges ${hyperEdgeFilter.replaceAll('record', 'edgeRecord')}
+    SORT '_key'
+    LIMIT ${input.page as number * limit}, ${limit}
+    RETURN {
+      'study': ${input.verbose === 'true' ? `(${verboseQuery})[0]` : 'edgeRecord._to'},
+      ${getDBReturnStatements(variantPhenotypeToStudy).replaceAll('record', 'edgeRecord')}
+    }
+    `
+  } else {
+    query = `
+      FOR record IN ${variantPhenotypeToStudy.db_collection_name as string}
+      FILTER ${hyperEdgeFilter}
+      SORT record._key
       LIMIT ${input.page as number * limit}, ${limit}
       RETURN {
-        'study': ${input.verbose === 'true' ? `(${verboseQuery})[0]` : 'edgeRecord._to'},
-        ${getDBReturnStatements(variantPhenotypeToStudy).replaceAll('record', 'edgeRecord')}
+        'study': ${input.verbose === 'true' ? `(${verboseQuery})[0]` : 'record._to'},
+        ${getDBReturnStatements(variantPhenotypeToStudy)}
       }
-      `
-    }
-  } else {
-    if (hyperEdgeFilter !== '') {
-      query = `
-        FOR record IN ${variantPhenotypeToStudy.db_collection_name}
-        FILTER ${hyperEdgeFilter}
-        SORT record._key
-        LIMIT ${input.page as number * limit}, ${limit}
-        RETURN {
-          'study': ${input.verbose === 'true' ? `(${verboseQuery})[0]` : 'record._to'},
-          ${getDBReturnStatements(variantPhenotypeToStudy)}
-        }
-      `
-    } else {
-      throw new TRPCError({
-        code: 'BAD_REQUEST',
-        message: 'At least one filter on variant or log10pvalue must be defined.'
-      })
-    }
+    `
   }
-
   return await ((await db.query(query)).all())
 }
 
 async function findPhenotypesFromVariantSearch (input: paramsFormatType): Promise<any[]> {
+  variantQueryValidation(input)
   delete input.organism
-  if (input.variant_id !== undefined) {
-    input._key = input.variant_id
-    delete input.variant_id
-  }
-
   if (input.funseq_description !== undefined) {
```

Comment:
This could be removed, we no longer have them in input?

Reply:
Will remove. Thanks!